src/modules/auth/auth.controller.ts

```diff
@@ -0,0 +1,94 @@
+import { Controller, Post, Body, Param, Get, Res, HttpStatus } from '@nestjs/common';
+import {
+	ApiBadRequestResponse,
+	ApiNotFoundResponse,
+	ApiOkResponse,
+	ApiOperation,
+	ApiParam,
+	ApiResponse,
+	ApiTags,
+	ApiUnauthorizedResponse,
+} from '@nestjs/swagger';
+import express from 'express';
+import { z } from 'zod';
+
+import { TranslateService } from '@modules/translate/translate.service';
+import { User } from '@modules/users/entities/user.entity';
+import { UsersDataService } from '@modules/users/services/users-data.service';
+import { validate } from '@utils/validate';
+
+import { AuthService } from './auth.service';
+import { UserPostDTO } from './dto/register.dto';
+import { UserSignInDTO } from './dto/sign-in.dto';
+import { TokenDTO } from './dto/token.dto';
+
+@ApiTags('Authentification')
+@Controller('auth')
+export class AuthController {
+	constructor(
+		private readonly t: TranslateService,
+		private readonly authService: AuthService,
+		private readonly userService: UsersDataService,
+	) {}
+
+	@Post('login')
+	@ApiOperation({ summary: 'Sign in a user with email and password' })
+	@ApiUnauthorizedResponse({ description: 'Unauthorized, password invalid' })
+	@ApiNotFoundResponse({ description: 'User not found' })
+	@ApiOkResponse({ description: 'OK', type: TokenDTO })
+	async login(@Body() signInDto: UserSignInDTO): Promise<TokenDTO> {
+		const schema = z
+			.object({
+				password: z.string(),
+				email: z.string().email(),
+			})
+			.strict();
+
+		validate(schema, signInDto);
+
+		return this.authService.signIn(signInDto.email, signInDto.password);
+	}
+
+	@Post('register')
+	@ApiOperation({ summary: 'Register a new user' })
+	@ApiOkResponse({ description: 'User created', type: User })
+	@ApiBadRequestResponse({ description: 'Bad request, invalid fields' })
+	async register(@Body() registerDto: UserPostDTO): Promise<User> {
+		const schema = z
+			.object({
+				password: z.string(),
+				first_name: z.string(),
+				last_name: z.string(),
+				email: z.string().email(),
+				birth_date: z.string().datetime(),
+			})
+			.strict();
+
+		validate(schema, registerDto);
+
+		return this.userService.register(registerDto);
+	}
+
+	@Get('confirm/:user_id/:token')
+	@ApiParam({ name: 'user_id', type: Number })
+	@ApiParam({ name: 'token', type: String })
+	@ApiOperation({ summary: 'Validate a user account and redirect after' })
+	@ApiNotFoundResponse({ description: 'User not found' })
+	@ApiBadRequestResponse({ description: 'Bad request, missing id/token or email already verified' })
+	@ApiUnauthorizedResponse({ description: 'Unauthorized, invalid token' })
+	@ApiResponse({
```

Comment:
It is the role of the front-end to make such redirections.

Reply:
Yes and no, here it's for email verification which are sent by the API and not the Sith, that's why I've put the redirect endpoint to bring back the users directly to the main website

The first thing I did was to put the redirect URL in the request param but it seamed unsafe to do such thing (GitHub CodeQL triggered on it), should I put it back ?

Reply:
It is not the role of the API to make redirections. Instead, the end-user should call the front, and it is the front whom is in charge of calling the API to validate the token. Then, the api will return a status code (200,400,...) and the front will manage the behaviour.